### DISCUSSION TITLE
TreeView.SetNodeStyling for Dynamic class value

### DISCUSSION THREAD

**Discussion by maglethong** [Megabit/Blazorise#3708] [Q&A]
Hi,

I am unsure if this is a desired behavior since the parameter has little official documentation, but I saw the 
`Action<TNode, Blazorise.TreeView.NodeStyling> NodeStyling` and immediately saw potential for some custom styling of nodes depending on an external condition, say, for example:
```
<TreeView Nodes = "RootElements"
          TNode= "NodeType"
          GetChildNodes= "GetChildNodes"
          HasChildNodes= "HasChildNodes"
          NodeStyling= "IsHighlight" >

@code{
    List<NodeType> HighlightedNodes { get; init; } = new ();
    private void IsHighlight(NodeType currentNode, NodeStyling nodeStyle) {
        nodeStyle.Class = HighlightedNodes.Contains(currentNode) ? "tree-item-highlight" : "";
    }
    // ...
}
```

Evidently `IsHighlight` gets called when the Tree refreshes, yet for previously rendered nodes, the class does not update.

is there a better way of achieving this or is that generally not possible?

I noticed [_TreeViewNodeContent](https://github.com/Megabit/Blazorise/blob/master/Source/Extensions/Blazorise.TreeView/_TreeViewNodeContent.razor.cs) seems to invoke the NodeStyling action only when building, so perhaps this is intended to work this way and There is a smarter path to the same result?

**Comment by stsrki** [comment]
It doesn't update for previously rendered nodes because that would slow down the rendering. No reason to update the parent node from its children nodes.

Maybe you could call StateHasChanged on the page level but I'm not sure if that could work.

**Reply by maglethong** [reply]
Sadly it does not
